### PR TITLE
Settings cleanup

### LINKED ISSUE #3658: Only show versions more recent than the one installed in Updates
We shouldn't show all the release notes when an admin has an update pending.

Also, if the instance is on a version that's newer than the latest, eg on a release candidate, we shouldn't notify them on an update _and_ show all previous releases.

### PR DESCRIPTION
**Pull request by tlrobinson**
- Cleanup settings code, remove some copypasta
- Consolidate map settings
- Move GeoJSON "add map" to modal
- Add a GeoJSON preview
- Do an actual ordered version comparison in updates panel. Resolves #3658

### CODE REVIEW

File: frontend/src/metabase/admin/settings/components/SettingsEmailForm.jsx [hunk 1]
```diff
@@ -184,10 +183,14 @@ export default class SettingsEmailForm extends Component {
             let errorMessage = (formErrors && formErrors.elements) ? formErrors.elements[element.key] : validationErrors[element.key];
             let value = formData[element.key] == null ? element.defaultValue : formData[element.key];
 
-            return <SettingsEmailFormElement
-                        key={element.key}
-                        element={{ ...element, value, errorMessage }}
-                        handleChangeEvent={this.handleChangeEvent.bind(this)} />
+            return (
+                <SettingsSetting
```
Comment (kdoh) [inline frontend/src/metabase/admin/settings/components/SettingsEmailForm.jsx]: SettingsSetting isn't my favorite component name in the world but to be fair I can't think of a better name.

File: frontend/src/metabase/admin/settings/components/widgets/CustomGeoJSONWidget.jsx [hunk 2]
```diff
@@ -0,0 +1,307 @@
...
+    }
+
+    static propTypes = {
+        setting: PropTypes.object.isRequired,
+        updateSetting: PropTypes.func.isRequired,
+        reloadSettings: PropTypes.func.isRequired
+    };
+    static defaultProps = {};
+
+    _saveMap = async (id, map) => {
```
Comment (kdoh) [inline frontend/src/metabase/admin/settings/components/widgets/CustomGeoJSONWidget.jsx]: I'm starting to get a bit confused about when we use `_` function names and when we don't (I do realize this is ported but I'm also curious). What's your current thinking on that?

Comment (tlrobinson) [inline frontend/src/metabase/admin/settings/components/widgets/CustomGeoJSONWidget.jsx]: The `_` prefix usually indicates a "private" method (only called from within the class), but you're right we're pretty inconsistent. In React you don't really call any methods on components from outside except for the lifecycle methods which are called by React itself, but I like including the underscore for consistency with non-React code, and as a little differentiator from the lifecycle methods.

This is the sort of thing we should write out in a code style guide.

**Comment by tlrobinson** [comment]
Alright this should be ready to go.

**Comment by kdoh** [comment]
:+1:

**Pull request merged**